core/src/main/java/io/aiven/klaw/auth/KwAuthenticationSuccessHandler.java

```diff
@@ -31,6 +31,9 @@ public class KwAuthenticationSuccessHandler extends SavedRequestAwareAuthenticat
   @Value("${klaw.ad.username.attribute:preferred_username}")
   private String preferredUsernameAttribute;
 
+  @Value("${klaw.ad.email.attribute:preferred_username}")
```

Comment:
preferred_username, and email are default user attributes in any provider. Azure AD, has preferred_username by default, and google has email by default

Reply:
I might be missing something but the below was my thinking

 @Value("${klaw.ad.username.attribute:preferred_username}")
  private String preferredUsernameAttribute;

@Value("${klaw.ad.email.attribute:preferred_username}")
private String emailAttribute;


So here preferredUsernameAttribute if not explicitly set in the application.properties will have the value preferred_username

emailAttribute if not explicitly set in the application.properties will have the value preferred_username as its value as well.

that means when we call getUserName the fallback value will be the same preffered_username
 UtilMethods.getUserName(authentication.getPrincipal(), "preferred_username", "preferred_username"));
 
 I think if we update emailAttribute to
 @Value("${klaw.ad.email.attribute:email}")
private String emailAttribute;


This will mean by default when we call getUsername we will be calling the below falling back on the value email instead of preferred_username.

UtilMethods.getUserName(authentication.getPrincipal(), "preferred_username", "email"));